source/Calamari.Testing/LogParser/ScriptOutputFilter.cs

```diff
@@ -225,20 +223,17 @@ public void Write(ProcessOutputSource source, string text)
 
         void WritePlainText(ProcessOutputSource source, string text)
         {
-            using (log.WithinBlock(logContext))
```

Comment:
I know this is only for test purposes, but I am not sure what impact removing this has?

Reply:
This is a good question, and what this line used to do, and why, was the reason it is going away :) What it used to do was change the thread static context to be the one with the correlationId (that's all ILogWithContext really was) before calling the callbacks. Those callback would have an ILog, most likely, that would depend on the thread context being right before they got called, and do terrible things if it wasn't. Now the tasklog instance has to be passed in, and it has the correlationId and there's no thread context involved, thus the callbacks can only write to the correct place.